### PR TITLE
CHEAP-644 Add mutations Vendor.updateAllMenuItemsStatus, Customer.createWalletTransaction and Customer.refundWallet

### PR DESCRIPTION
**Pull request by bholagabbar**
Screenshot from 2019-04-06 02-11-49

### CODE REVIEW

File: src/app/controllers/CustomerController.js [hunk 1]
```diff
@@ -282,8 +284,59 @@ class CustomerController {
...
+     * @param {string} id - The id of the Customer
+     * @param  {int} vendor_id - ID of the Vendor issuing the refund
+     * @param  {string} amount - The amount to refund the wallet (in cents)
+     * @param  {string} order_id - Optional orderId selected payment method
+     * @returns {Promise<any>} - The id of the wallet that was reloaded
+     */
+    refundWallet(id, vendor_id, amount, order_id=null) {
+        return new Promise((resolve, reject) => {
+            let mutationString = `
+                mutation ($id:String!, $vendor_id:String!, $amount:Int!, $order_id:String) {
```
Comment (ralph-dev) [inline src/app/controllers/CustomerController.js]:
```suggestion
                mutation ($id: String!, $vendor_id: String!, $amount: Int!, $order_id: String) {
```

File: src/app/controllers/CustomerController.js [hunk 2]
```diff
@@ -282,8 +284,59 @@ class CustomerController {
...
+                    refundCustomerWallet(id:$id, vendor_id:$vendor_id, amount:$amount, order_id:$order_id) {
```
Comment (ralph-dev) [inline src/app/controllers/CustomerController.js]:
```suggestion
                    refundCustomerWallet(id: $id, vendor_id: $vendor_id, amount: $amount, order_id: $order_id) {
```

File: src/app/controllers/CustomerController.js [hunk 3]
```diff
@@ -282,8 +284,59 @@ class CustomerController {
...
+     * @param  {int} transaction_type - Transaction type, either 'reload' or 'purchase'
+     * @param  {string} amount - The amount in cents
+     * @param  {string} description - Optional description for transaction
+     * @returns {Promise<any>} - The id of the wallet that was reloaded
+     */
+    createWalletTransaction(id, transaction_type, amount, description=null) {
+        return new Promise((resolve, reject) => {
+            let mutationString = `
+                mutation ($id:String!, $transaction_type:String!, $amount:Int!, $description:String) {
+                    createCustomerWalletTransaction(id:$id, transaction_type:$transaction_type, amount:$amount, description:$description) {
```
Comment (ralph-dev) [inline src/app/controllers/CustomerController.js]:
```suggestion
                    createCustomerWalletTransaction(id: $id, transaction_type: $transaction_type, amount: $amount, description: $description) {
```

File: src/app/controllers/CustomerController.js [hunk 4]
```diff
@@ -282,8 +284,59 @@ class CustomerController {
...
+     * @param {string} id - The id of the Customer
+     * @param  {int} transaction_type - Transaction type, either 'reload' or 'purchase'
+     * @param  {string} amount - The amount in cents
+     * @param  {string} description - Optional description for transaction
+     * @returns {Promise<any>} - The id of the wallet that was reloaded
+     */
+    createWalletTransaction(id, transaction_type, amount, description=null) {
+        return new Promise((resolve, reject) => {
+            let mutationString = `
+                mutation ($id:String!, $transaction_type:String!, $amount:Int!, $description:String) {
```
Comment (ralph-dev) [inline src/app/controllers/CustomerController.js]:
```suggestion
                mutation ($id: String!, $transaction_type: String!, $amount: Int!, $description: String) {
```

**Review by ralph-dev — APPROVED** [review]
LGTM

File: src/app/controllers/CustomerController.js (hunk 1 above)
Comment (bholagabbar) [reply src/app/controllers/CustomerController.js]: Jeez :P

**Pull request merged**